onnxruntime/core/optimizer/initializer.cc

```diff
@@ -54,6 +70,44 @@ Status Initializer::ReadExternalRawData(
 
   return Status::OK();
 }
+
+Status Initializer::ReadExternalRawData(
+    const ONNX_NAMESPACE::TensorProto& tensor_proto, const std::unordered_map<std::string, const void*>& external_data_map, std::vector<char>& raw_data) {
+  ORT_RETURN_IF_NOT(
+      tensor_proto.data_type() != ONNX_NAMESPACE::TensorProto_DataType_UNDEFINED &&
+          tensor_proto.data_type() != ONNX_NAMESPACE::TensorProto_DataType_STRING,
+      "External data type must not be UNDEFINED or STRING.");
+
+  std::unique_ptr<ExternalDataInfo> external_data{};
+  ORT_RETURN_IF_ERROR(ExternalDataInfo::Create(tensor_proto.external_data(), external_data));
+
+  size_t actual_tensor_data_length;
+  ORT_RETURN_IF_ERROR(utils::GetSizeInBytesFromTensorProto<0>(
+      tensor_proto, &actual_tensor_data_length));
+  const size_t external_data_length = external_data->GetLength();
+
+  ORT_RETURN_IF_NOT(
+      external_data_length == 0 ||
+          external_data_length == actual_tensor_data_length,
+      "TensorProto external data size mismatch. ",
+      "Computed size: ", actual_tensor_data_length,
+      ", external_data.length: ", external_data_length);
+
+  auto buffer_offset = external_data->GetOffset();
+
+  auto external_data_name = external_data->GetRelPath();
+  auto external_data_key = ONNXStringToString(external_data_name);
+  auto it = external_data_map.find(external_data_key);
+  ORT_RETURN_IF(it == external_data_map.end(), "Cannot find the key (" + external_data_key + ") in the external_data_map");
+
+  std::vector<char> buffer(actual_tensor_data_length);
```

Comment:
>buffer [](http://example.com/codeflow?start=20&length=6)

This needless initializers buffer with zeros which are thrown away, but shows up on perf profiler. reserve() + push_back()